authorization/v2/authorizer.go

```diff
@@ -116,7 +124,24 @@ func (a *authorizer) GetAuthorizedResources(token string, urisToCheck []common.U
 		return nil, errors.Wrap(common.ErrInvalidToken, err.Error())
 	}
 
-	claims.AllowedResources, err = a.filterUrisWithInvalidMetadata(claims.AllowedResources)
+	var claimedResources common.UniformResourceIdentifiers
+	if claims.TokenType == User {
+		user, err := a.userService.GetUserWithID(ctx, claims.Id)
+		if err != nil {
+			return nil, err
+		}
+
+		rolePermissions, err := a.cfg.UserRole.GetRolePermissions(string(user.Role))
```

Comment:
Could you update the `GetRolePermissions` function to accept a `UserRole` rather than a `string`. Because inside `GetRolePermissions` we're just casting the `string` to a `UserRole` anyway.

Reply:
Yuup 👍